runtime/src/bank.rs

```diff
@@ -6878,6 +6878,22 @@ impl TransactionProcessingCallback for Bank {
             .map(|(stake, _)| (*stake))
             .unwrap_or(0)
     }
+
+    fn calculate_fee(
+        &self,
```

Comment:
why is this implemented on `Bank`? we use none of its fields

Reply:
The bank already implements the callback trait, so adding another callback was simpler. We need a trait or a callback to decouple solana-fee from SVM. 

What's your suggestion? Happy to rework it.

Reply:
sounds like this trait is just a kitchen sink. not well-defined api. i'll try to find time to look at it